test/tests/routing.test.js

```diff
@@ -86,10 +124,65 @@ describe('Handling custom endpoint routes', () => {
 
   after(() => tapestry.server.stop())
 
-  it('Route matched, custom endpoint works', (done) => {
-    request.get(`${uri}/basic-endpoint`, (err, res, body) => {
+  it('Route matched, string endpoint works', (done) => {
+    request.get(`${uri}/string-endpoint`, (err, res, body) => {
       expect(body).to.contain(prepareJson(dataPages.data))
       done()
     })
   })
+
+  it('Route matched, array endpoint works', (done) => {
+    request.get(`${uri}/array-endpoint`, (err, res, body) => {
+      const expectedJson = `[{"data":[${
+        prepareJson(dataPages.data)
+      },${
+        prepareJson(dataPosts.data)
+      }]}]`
+      expect(body).to.contain(expectedJson)
+      done()
+    })
+  })
+
+  it('Route matched, object endpoint works', (done) => {
+    request.get(`${uri}/object-endpoint`, (err, res, body) => {
+      const expectedJson = `[{"data":{"pages":${
+        prepareJson(dataPages.data)
+      },"posts":${
+        prepareJson(dataPosts.data)
+      }}}]`
```

Comment:
Small stylistic thing – would it be easier to write/format this if you used `JSON.stringify`? Happy to leave of course not a biggie